StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1210UsingDirectivesMustBeOrderedAlphabeticallyByNamespace.cs

```diff
@@ -50,108 +56,60 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
-                {
-                    continue;
-                }
+        private void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var namespaceDeclaration = (NamespaceDeclarationSyntax)context.Node;
 
-                string precedingFullyQualifiedNamespace = precedingNamespaceSymbol.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
-                if (string.Compare(fullyQualifiedNamespace, precedingFullyQualifiedNamespace, StringComparison.OrdinalIgnoreCase) >= 0)
-                {
-                    continue;
-                }
+            ProcessUsings(namespaceDeclaration.Usings, context);
+        }
 
-                string @namespace = namespaceSymbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
-                string precedingNamespace = precedingNamespaceSymbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
+        private static void ProcessUsings(SyntaxList<UsingDirectiveSyntax> usings, SyntaxNodeAnalysisContext context)
+        {
+            var usingsDirectivesWithoutAliasAndStatic = usings.Where(ExcludeAliasAndStaticUsingDirectives);
+            var systemUsingDirectives = usingsDirectivesWithoutAliasAndStatic.Where(directive => directive.IsSystemUsingDirective());
+            var usingDirectives = usingsDirectivesWithoutAliasAndStatic.Where(directive => !directive.IsSystemUsingDirective());
 
-                // Using directive for '{namespace}' must appear before directive for '{precedingNamespace}'
-                context.ReportDiagnostic(Diagnostic.Create(Descriptor, syntax.GetLocation(), @namespace, precedingNamespace));
-                break;
-            }
+            CheckOrderAndReportDiagnostic(usingDirectives.ToList(), context);
+            CheckOrderAndReportDiagnostic(systemUsingDirectives.ToList(), context);
         }
 
-        private static string GetTopLevelNamespace(SemanticModel semanticModel, UsingDirectiveSyntax syntax, out INamespaceSymbol namespaceSymbol, CancellationToken cancellationToken)
+        private static void CheckOrderAndReportDiagnostic(IList<UsingDirectiveSyntax> usings, SyntaxNodeAnalysisContext context)
         {
-            SymbolInfo symbolInfo = semanticModel.GetSymbolInfo(syntax.Name, cancellationToken);
-            namespaceSymbol = symbolInfo.Symbol as INamespaceSymbol;
-            if (namespaceSymbol == null)
+            if (usings.Count <= 1)
             {
-                return null;
+                return;
             }
 
-            string fullyQualifiedName = namespaceSymbol.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
-            string name = fullyQualifiedName;
+            var properPositionsOfUsingDirectives = usings.OrderBy(directive => directive.Name.ToUnaliasedString()).Select((directive, idx) => new { Directive = directive, Index = idx }).ToDictionary(item => item.Directive, item => item.Index);
```

Comment:
Makes sense.
It looks like StyleCop gives generic message for every using directive that following directive is less than:

```csharp
    using System.Threading;
    using System.Linq;
    using System.IO;
    using System.Threading.Tasks;
    using System.Security;
```
For that example StyleCop generates message for `System.Threading`, `System.Linq`, `System.Threading.Tasks`.

Just to clarify, should we go the same?